### PR TITLE
Fix: #3929

### PR DESCRIPTION
**Pull request by HHongSeungWoo**
## Description

I have fixed issue #3929

## Related Issue

#3929

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] I have added a storybook story or extended an existing story to show my changes
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/chart/generateCategoricalChart.tsx [hunk 1]
```diff
@@ -233,7 +233,8 @@ const getTooltipContent = (
      * Defaulting to chartData below to fix an edge case where the tooltip does not include data from all charts
      * when a separate dataset is passed to chart prop data and specified on Line/Area/etc prop data
      */
-    const { data = chartData } = child.props;
+    const data = (child.props.data ?? chartData).slice(state.dataStartIndex, state.dataEndIndex + 1);
+
```
Comment (andrewangelle) [inline src/chart/generateCategoricalChart.tsx:237]: Nice 👍

**Review by nikolasrieble — APPROVED** [review]
🚀 Thank you for the fix

**Comment by bjarn** [comment]
Hi @nikolasrieble / @HHongSeungWoo, I think this change introduced an issue with tooltips as 2.9.2 doesn't have this issue.

```
generateCategoricalChart.js:185 Uncaught TypeError: Cannot read properties of undefined (reading 'slice')
    at eval (webpack-internal:///(app-pages-browser)/./node_modules/recharts/es6/chart/generateCategoricalChart.js:244:130)
    at Array.reduce (<anonymous>)
    at getTooltipContent (webpack-internal:///(app-pages-browser)/./node_modules/recharts/es6/chart/generateCategoricalChart.js:232:25)
    at getTooltipData (webpack-internal:///(app-pages-browser)/./node_modules/recharts/es6/chart/generateCategoricalChart.js:280:25)
    at CategoricalChartWrapper.getMouseInfo (webpack-internal:///(app-pages-browser)/./node_modules/recharts/es6/chart/generateCategoricalChart.js:1633:27)
    at CategoricalChartWrapper.eval [as triggeredAfterMouseMove] (webpack-internal:///(app-pages-browser)/./node_modules/recharts/es6/chart/generateCategoricalChart.js:966:27)
    at eval (webpack-internal:///(app-pages-browser)/./node_modules/recharts/es6/chart/generateCategoricalChart.js:1016:15)
    at HTMLUnknownElement.callCallback (webpack-internal:///(app-pages-browser)/./node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js:20458:14)
    at Object.invokeGuardedCallbackImpl (webpack-internal:///(app-pages-browser)/./node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js:20507:16)
    at invokeGuardedCallback (webpack-internal:///(app-pages-browser)/./node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js:20582:29)
    at invokeGuardedCallbackAndCatchFirstError (webpack-internal:///(app-pages-browser)/./node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js:20596:25)
    at executeDispatch (webpack-internal:///(app-pages-browser)/./node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js:31933:3)
    at processDispatchQueueItemsInOrder (webpack-internal:///(app-pages-browser)/./node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js:31965:7)
    at processDispatchQueue (webpack-internal:///(app-pages-browser)/./node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js:31978:5)
    at dispatchEventsForPlugins (webpack-internal:///(app-pages-browser)/./node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js:31989:3)
    at eval (webpack-internal:///(app-pages-browser)/./node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js:32179:12)
    at batchedUpdates$1 (webpack-internal:///(app-pages-browser)/./node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js:24790:12)
    at batchedUpdates (webpack-internal:///(app-pages-browser)/./node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js:28650:12)
    at dispatchEventForPluginEventSystem (webpack-internal:///(app-pages-browser)/./node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js:32178:3)
    at dispatchEvent (webpack-internal:///(app-pages-browser)/./node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js:29946:5)
    at dispatchContinuousEvent (webpack-internal:///(app-pages-browser)/./node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js:29931:5)
```

**Comment by HHongSeungWoo** [comment]
@bjarn Can you provide a minimal reproducible example??

**Comment by bjarn** [comment]
> @bjarn Can you provide a minimal reproducible example??

You're quick 🚀

I'll provide you with one in a couple of hours after work. Thanks!
